### ISSUE TITLE
Using the created Nginx & Letsencrypt companion for SSL certs on a different server/IP address

### ISSUE THREAD

**Issue by trueself-belief** [8gears/containerized-guacamole#10]
This may be out of scope for here but wanted to ask as i can't seem to get it. 
I have this fully working which is great. 

I am now looking to use the nginx + letsencrypt containers to help secure an SSL cert for a webserver on a different IP address. 

Is this possible? 

(I have set up a 'webserver.conf' file in /etc/nginx.conf.d/ - and this is proxying to the correct IP and working - without SSL)

I see how to do it if I wanted to create another docker container within the same server+network but can't figure out how to do it with an existing server on a different IP address!

**Comment by Vad1mo** [comment]
Yes, this is possible, and it is a plain nginx-proxy feature rather than

nginx-proxy only generates a vhost for a **container** that declares
`VIRTUAL_HOST`, which is why a hand-written file in `conf.d` gets you routing
but no certificate. You need two pieces:

**1. A placeholder container to declare the hostname**, so docker-gen emits the
vhost and acme-companion requests a certificate for it. It does not have to
serve anything:

```yaml
  ext-placeholder:
    image: alpine
    command: sleep infinity
    environment:
      VIRTUAL_HOST: other.example.com
      LETSENCRYPT_HOST: other.example.com
```

**2. A location override pointing at the other machine.** nginx-proxy includes
`/etc/nginx/vhost.d/<domain>_location_override` in place of the generated
`location` block, so the file must contain a complete block, not just a
`proxy_pass` line. That was the part that tripped me up first:

```nginx
location / {
    proxy_pass http://192.0.2.50;
}
```

Write it into the `vhost.d` volume and reload.

**Verified.** With a web server that is *not* a proxy-managed container on
`172.31.0.5`, a request to the proxy for that hostname came back from the
external machine:

```
$ curl -H 'Host: ext.example' http://proxy/
HELLO-FROM-EXTERNAL-SERVER
```

Two things worth knowing before you build on this:

* DNS for `other.example.com` must point at **this** host, because the HTTP-01
  challenge is served by this nginx. TLS is terminated here and the hop to the
  other machine is plain HTTP over your network, so that leg should be a link
  you trust. The other server never gets a certificate of its own.
* If the other machine must present its own certificate, this is the wrong
  tool. Run acme.sh or certbot there directly.

Leaving this open as a reference since it comes up repeatedly, and the
`_location_override` needing a full `location` block is not obvious.
